### PR TITLE
Add date-sorting to watch() when no IDs are given

### PR DESCRIPTION
**Pull request by EmRowlands**
When no video IDs are selected, videos are sorted by date, rather than ID.
When videos are listed, they are also listed by date.

### CODE REVIEW

**Comment by woefe** [comment]
I don't like this being enabled by default. The videos are actually sorted right now. They are sorted by channel and within a channel by date.
See:
https://github.com/woefe/ytcc/blob/5542be08524d38f6fc86feec345b2a9410a66fe8/ytcc/database.py#L196

Global sorting by date should be optional.

**Comment by EmRowlands** [comment]
I sorted the videos outside of the SQL so that it only came into effect at specific points. That's fair enough about making it non-default, it would probably make most sense to enable it using a config file option, rather than using a command line option.

**Comment by EmRowlands** [comment]
I have changed to allow for sorting on various parameters defined in the dictionary:

https://github.com/EmRowlands/ytcc/blob/1ec12c62917d94d97376f93bc16f83fe966fc611/ytcc/database.py#L183-L191

The only thing that is missing from this is the translation for this line, but I don't know German, nor do I particularly trust online translators.

https://github.com/EmRowlands/ytcc/blob/1ec12c62917d94d97376f93bc16f83fe966fc611/ytcc/database.py#L201

File: ytcc/database.py [hunk 1]
```diff
@@ -172,11 +173,36 @@ def get_videos(self, channel_filter: Optional[List[str]] = None, begin_timestamp
...
+        
+        # If no argument is passed, use a default.
+        sort_params : str = "c.id, v.publish_date"
+        if orderby:
+            sort_list : List[str] = list()
+            for col in orderby:
+                # Take care of extra whitespace
+                col=col.strip()
+                if not col in config_to_query_param.keys():
+                    print(_("Ignoring unknown orderBy option: {col}").format(col=col))
```
[the remark this answers is not in the document]
Comment (EmRowlands) [reply ytcc/database.py]: I've changed to using exceptions, and catching them in cli.py to print an error.

**Comment by woefe** [comment]
I rewrote the entire database module. It uses SQLAlchemy now. No more custom SQL statements (almost). SQLAlchemy made it pretty simple to implement sorting. It's in v1.8.0.

**Pull request closed without merging**